### PR TITLE
FunExt implies WeakFunExt

### PR DESCRIPTION
**Pull request by MatthiasHu**
Part of https://github.com/rzk-lang/sHoTT/issues/73.

### CODE REVIEW

**Comment by MatthiasHu** [comment]
This is only a draft pull request for now. I am grateful for any comments/corrections of course!

File: src/hott/06-contractible.rzk.md [hunk 1]
```diff
@@ -407,27 +407,48 @@ separate hypothesis.
...
+  (f : (a : A) → is-contr (C a))
+  : (a : A) → C a
+  :=
+  \ a → first (f a)
+
+#def weakfunext-funext
+  (funext : FunExt)
+  : WeakFunExt
+  :=
+  ( \ A → \ C → \ f →
```
Comment (emilyriehl) [inline src/hott/06-contractible.rzk.md:443]: Here you don't need the opening parentheses (after the :=) and can collect all the lambdas as `\ A C f →`

File: src/hott/06-contractible.rzk.md [hunk 2]
```diff
@@ -407,27 +407,48 @@ separate hypothesis.
...
+    ( map-weakfunext-funext A C f ,
+      ( \ (g : (a : A) → C a) →
```
Comment (emilyriehl) [inline src/hott/06-contractible.rzk.md:445]: We don't often give type annotations here (for your `\ g`) but you could add them if you think they help.

File: src/hott/06-contractible.rzk.md [hunk 3]
```diff
@@ -407,27 +407,48 @@ separate hypothesis.
 #def WeakFunExt : U
   :=
     ( A : U ) → (C : A → U) →
+    -- TODO: rename f to is-contr-C (also below)?
```
Comment (emilyriehl) [inline src/hott/06-contractible.rzk.md:410]: I like your suggestion to use `is-contr-C`.

**Review by emilyriehl — COMMENTED** [review]
This looks good. If you want to clean it up (deleting the todos and the hole) we could merge this part now and you could start a new PR for the rest. But in that case you should edit your comment above to say "part of #74" rather than "closes" because I think github might automatically close the issue otherwise.

File: src/hott/06-contractible.rzk.md (hunk 2 above)
Comment (MatthiasHu) [reply src/hott/06-contractible.rzk.md]: Ah, I think I meant to remove it again, did that now, thanks!

**Comment by MatthiasHu** [comment]
This is ready now.

Is the name `map-for-weakfunext` ok?

**Comment by TashiWalde** [comment]
On a tangentially related note, I think that "unique unique choice" is a better name than "weak function extensionality" since it is much more descriptive. Not sure if its worth changing at this point, just throwing it out there...

**Comment by jonweinb** [comment]
> This is ready now.
> 
> Is the name `map-for-weakfunext` ok?

I'd go for `map-for-weakfunext` according to previous conventions. Otherwise, I can merge.

**Comment by MatthiasHu** [comment]
Sorry -- are you saying the name should be changed, or are you saying it can stay as it is?

**Comment by jonweinb** [comment]
Oops, sorry for the typo. I think `map-weakfunext` is more in line with preexisting conventions than `map-for-weakfunext`, so arguing *for* a change.

**Comment by MatthiasHu** [comment]
Thanks, I changed it.

**Review by jonweinb — APPROVED** [review]

**Pull request merged**